rllib/algorithms/algorithm.py

```diff
@@ -3890,11 +3890,12 @@ def _compile_iteration_results(self, *, train_results, eval_results):
 
             # Return dict (shallow copy of `train_results`).
             results: ResultDict = train_results.copy()
-            # Backward compatibility `NUM_ENV_STEPS_SAMPLED_LIFETIME` is now:
-            # `ENV_RUNNER_RESULTS/NUM_ENV_STEPS_SAMPLED_LIFETIME`.
-            results[NUM_ENV_STEPS_SAMPLED_LIFETIME] = results.get(
-                ENV_RUNNER_RESULTS, {}
-            ).get(NUM_ENV_STEPS_SAMPLED_LIFETIME, 0)
+            if results.get(NUM_ENV_STEPS_SAMPLED_LIFETIME) is None:
+                # Backward compatibility `NUM_ENV_STEPS_SAMPLED_LIFETIME` is now:
+                # `ENV_RUNNER_RESULTS/NUM_ENV_STEPS_SAMPLED_LIFETIME`.
+                results[NUM_ENV_STEPS_SAMPLED_LIFETIME] = results.get(
+                    ENV_RUNNER_RESULTS, {}
+                ).get(NUM_ENV_STEPS_SAMPLED_LIFETIME, 0)
```

Comment:
No need to always set this from env runner results.
Specifically, with BC, there are no env runner results but we set NUM_ENV_STEPS_SAMPLED_LIFETIME in the training step function.

Reply:
I would add it as comment in the code, I was looking for this exact info!